okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -1208,10 +1213,17 @@ public void rstStream(int streamId, ErrorCode errorCode) {
       Status status = toGrpcStatus(errorCode).augmentDescription("Rst Stream");
       boolean stopDelivery =
           (status.getCode() == Code.CANCELLED || status.getCode() == Code.DEADLINE_EXCEEDED);
-      finishStream(
-          streamId, status,
-          errorCode == ErrorCode.REFUSED_STREAM ? RpcProgress.REFUSED : RpcProgress.PROCESSED,
-          stopDelivery, null, null);
+      synchronized (lock) {
```

Comment:
Maybe create the event in the finishStream() method?

Reply:
I saw that, but from the spot you startTask(), it looks like the only job we care about in the resStream() is finishStream(). If it is the case, why not startTask() in the finishStream() method.

Reply:
`finishiStream()` will also be called in other places (e.g., cancel, handle frame error, etc), which we do not want to trace. Here we are tracing the event for receiving `RST_STREAM` specifically.